consensus/src/chained_bft/event_processor.rs

```diff
@@ -470,16 +471,18 @@ impl<T: Payload> EventProcessor<T> {
         self.safety_rules.update(qc);
 
         let mut highest_committed_proposal_round = None;
-        if let Some(new_commit) = qc.committed_block_id() {
-            if let Some(block) = self.block_store.get_block(new_commit) {
-                let finality_proof = qc.ledger_info().clone();
-                // We don't want to use NIL commits for pacemaker round interval calculations.
-                if !block.is_nil_block() {
-                    highest_committed_proposal_round = Some(block.round());
-                }
-                self.process_commit(block, finality_proof).await;
+        if let Some(block) = qc
+            .committed_block_id()
+            .and_then(|new_commit| self.block_store.get_block(new_commit))
```

Comment:
if let ... if let vs if let ... .and_then.  These are equivalent.  I don't have a strong preference here, rather that we just pick a way and make it consistent throughout the codebase.

Reply:
They're not quite equivalent in general (`and_then` is a more specific / less general construct), and it's not always possible to obtain this consistency ( for example here the inner `if let` is required by concurrency concerns ...). 

To scout for the more narrow use of an outer `if let` that doesn't quite pay for its genericity, I found them all via a simple:
```
fga@fga-fedora-PC0Z3JY1➜tmp/calibra/consensus(safety_refactorings✗)» find ./ -iname '*.rs' -exec perl -lne 'undef $/; print $ARGV, "\n", $1 while (m/((?: if let Some\([^\n]+\{)\n *if let Some\(.*)/g)' '{}' \;                                                       [7:56:09]
./src/chained_bft/block_storage/block_tree.rs
 if let Some(block) = self.try_get_block(&block_id) {
                if let Some(time_to_qc) = duration_since_epoch()
./src/chained_bft/safety/safety_rules_test.rs
 if let Some(children) = children_table.get(&query.0) {
            if let Some((depth, mut subchain)) = children.iter().cloned().flat_map(|(child_id, child_round)|
./src/chained_bft/persistent_storage.rs
 if let Some(committed_block_id) = qc.committed_block_id() {
                    if let Some(round) = id_to_round.get(&committed_block_id) {
```
and the last two don't need this conversion:
- in `safety_rules_test` because there's an `else` to the outer `if`
- in `persistent_storage` because the inner scope requires both `committed_block_id` and `round` 

I added a fix of the first one to this PR

Reply:
Thanks for the explanation, agree that *.and_then is not always possible.  I'm okay with the .and_then construction in these places.